src/reducers/scrollStateHelper.js

```diff
@@ -252,8 +252,10 @@ export function updateRowCount(state, { rowsCount }) {
   state.scrollContentHeight = rowsCount * state.rowHeight;
   state.rowOffsets = PrefixIntervalTree.uniform(rowsCount, state.rowHeight);
   state.storedHeights = new Array(rowsCount);
+  state.rows = new Array(rowsCount);
   for (var i = 0; i < rowsCount; i++) {
     state.storedHeights[i] = state.rowHeight;
+    state.rows[i] = i;
```

Comment:
Hmm can you dig into this a bit more and see if there's an in between that doesn't suffer from the issues Kam pointed out?

If you hit a wall on it after looking for a few hours, feel free to schedule a time with me to talk through the problem.